### PR TITLE
libobs-opengl: Request 3.2 compatibility profile 

### PR DESCRIPTION
**Pull request by admshao**
We already verify for the minimum 2.1 version after the context
creation so this will let the driver pick the highest version
available.

With this change on Nvidia proprietary it finally gives me the highest
context available. Mesa users were already getting the highest version
as we could see on user logs.

This PR does not touch gl-cocoa.m because on Apple things are different.
If we remove the request for 3.2 core profile it will fallback to 2.1 and i
don't know how to proper check for 4.1 (Max apple will support)

### CODE REVIEW

**Comment by computerquip-streamlabs** [comment]
This is incorrect and there's nothing in the GLX standard that says the highest profile will be chosen if left to a default. If I remember correctly, this is the case for Mesa. 

Is there a reason you want a higher version of OpenGL selected?

**Comment by admshao** [comment]
> This is incorrect and there's nothing in the GLX standard that says the highest profile will be chosen if left to a default.

There were no word of that in the spec. Correct. That was the result from tests on nvidia proprietary + mesa git.

> If I remember correctly, this is the case for Mesa.

Mesa is already returning the latest version available with the GLX_CONTEXT_CORE_PROFILE_BIT_ARB and that motivated me to start looking why NVIDIA wasn't.

> Is there a reason you want a higher version of OpenGL selected?

It would allow the obs graphics subsystem to be expanded since there are a growing number of plugins that do crazy stuff with the DX subsystem in windows and could be ported to GL subsystem. Higher profile version while still conformant to 3.2 would allow new extensions to be used to the same graphics subsystem and a really minor and maybe not valid point is that we don't limit the context on windows (i know opengl renderer there is deprecated).

I updated the PR to change the GLX_CONTEXT_CORE_PROFILE_BIT_ARB to GLX_CONTEXT_COMPATIBILITY_PROFILE_BIT_ARB to better comply with the only words in the spec i could find.

" If version 3.2 or greater is requested, the context returned may implement any of the following versions:
      * The requested profile of the requested version.
      * The requested profile of any later version, so long as no
        features have been removed from that later version and profile. "

Thanks for the review.

**Comment by computerquip-streamlabs** [comment]
I meant to look into this further but haven't had the time. What I can say without much research is that you can probably get away with detecting GL extensions (I think libobs exposes the GLAD wrapper headings as well) and using those dynamically rather than trying to raise the required version of OpenGL. 

Take this with a grain of salt since my memory has faded a bit on the topic: An OpenGL implementation that supports version 3.2 is required to support the core profile bit.  If 3.2 is requested, a valid OpenGL implementation may return 3.2 or anything that implements 3.2. 

The problem with compatibility mode is that ARB_compatibility is optional. Mesa takes advantage of this, there is no compatibility bit capability in Mesa. The compatibility mode they provide is for 3.1 (sometimes) and below only.

In other words, the only thing you can expect with OpenGL context versioning is what you ask for is what you should expect. If you ask for 3.2 core, you should expect 3.2 core, nothing more, nothing less. You can test for the version that it actually provides you and you can base things on that but you should never make the assumption that all drivers are going to have similar behavior when it's not defined by standard to have that behavior. 

Some of that might have also changed since I last looked into as well. I think there was some ARB_compatibility work back in 2017 at some point but don't know if it ever actually made it in.

**Comment by admshao** [comment]
Thanks for the review again.
Detecting GL extensions dynamically probably is the best way to handle this.
